lib/Frontend/CompilerInvocation.cpp

```diff
@@ -633,6 +646,37 @@ static bool ParseLangArgs(LangOptions &Opts, ArgList &Args,
                      A->getAsString(Args), A->getValue());
     }
   }
+  
+  // Check if the locale code is supported.
+  if (Arg *A = Args.getLastArg(OPT_locale)) {
+    std::string localeCode = A->getValue();
+
+    if (llvm::any_of(supportedLocaleCodes, [&](const char *locale) {
+          return localeCode != locale;
+        })) {
+      std::string validLocaleCodes = "";
+      llvm::interleave(
+          std::begin(supportedLocaleCodes), std::end(supportedLocaleCodes),
+          [&](std::string locale) { validLocaleCodes += locale; },
+          [&] { validLocaleCodes += ", "; });
+
+      // Incorrect locale code shouldn't crash the compiler,
+      // and the compiler will fallback to the default messages.
+      Diags.diagnose(SourceLoc(), diag::error_invalid_locale_code,
+                     validLocaleCodes);
+    } else {
+      // Check if the localization path exists but it doesn't have a file
```

Comment:
I don't know if this is the correct way to diagnose if the localization path exists but it doesn't have a file for the specified locale.

Reply:
Maybe all the validations should be separated in a specific function like `validateLocatizationArgs` ... this will be consistent with what we have in some other places like for example https://github.com/apple/swift/blob/8a39f82677b75c5f617c97321fda08e85b915c4a/lib/Driver/Driver.cpp#L208
What do you think @xedin ?

Reply:
Sure, we could do that. I'm kind of wondering whether we shouldn't actually explicitly add `supportedLocales` but instead just see what we have in terms of translation files and build a list from that...